src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,239 @@
+
+            if (storedValues.TargetFrameworkMoniker != null && storedValues.TargetFrameworkMoniker.Contains("Unsupported"))
+            {
+                return await base.OnGetUnevaluatedPropertyValueAsync(propertyName, "", defaultProperties);
+            }
+
+            return await base.OnGetUnevaluatedPropertyValueAsync(propertyName, ExtractValue(storedValues, propertyName) ?? "", defaultProperties);
+        }
+
+        private static async Task<ComplexTargetFramework> GetStoredComplexTargetFramework(ConfigurationGeneral configuration)
+        {
+            ComplexTargetFramework storedValues = new ComplexTargetFramework
+            {
+                TargetFrameworkMoniker = (string?)await configuration.TargetFrameworkMoniker.GetValueAsync(),
+                TargetPlatformIdentifier = (string?)await configuration.TargetPlatformIdentifier.GetValueAsync(),
+                TargetPlatformVersion = (string?)await configuration.TargetPlatformVersion.GetValueAsync(),
+                TargetFramework = (string?)await configuration.TargetFramework.GetValueAsync()
+            };
+
+            return storedValues;
+        }
+
+        /// <summary>
+        /// Extracts the specified value from the ComplexTargetFramework structure.
+        /// </summary>
+        /// <param name="values">The strucutre were all the values are.</param>
+        /// <param name="propertyName">The property's value we want to return.</param>
+        /// <returns></returns>
+        private static string? ExtractValue(ComplexTargetFramework values, string propertyName)
+        {
+            return propertyName switch
+            {
+                InterceptedTargetFrameworkProperty => values.TargetFrameworkMoniker,
+                TargetPlatformProperty => values.TargetPlatformIdentifier,
+                TargetPlatformVersionProperty => values.TargetPlatformVersion,
+                _ => null,
+            };
+        }
+
+        /// <summary>
+        /// Combines the three values that make up the TargetFramework property.
+        /// I.e. { net5.0, windows, 10.0 } => net5.0-windows10.0
+        /// { net5.0, null, null } => net5.0
+        /// { null, null, null } => String.Empty
+        /// </summary>
+        /// <param name="complexTargetFramework">A struct with each property.</param>
+        /// <returns>A string with the combined values.</returns>
+        private async Task<string> ComputeValueAsync(ComplexTargetFramework complexTargetFramework)
+        {
+            if (!Strings.IsNullOrEmpty(complexTargetFramework.TargetFrameworkMoniker))
+            {
+                string targetFrameworkAlias = await GetTargetFrameworkAlias(complexTargetFramework.TargetFrameworkMoniker);
+
+                if (string.IsNullOrEmpty(targetFrameworkAlias))
+                {
+                    return string.Empty;
+                }
+
+                if (targetFrameworkAlias.Contains("Unsupported"))
```

Comment:
Where does this `"Unsupported"` value come from? I don't remember seeing this in the SDK's items. It looks like `GetTargetFrameworkAlias` returns `""` for unknown values.

Reply:
True, this validation was used before I added it in the `GetTargetFrameworkAlias`, where I could get something like this: 
![image](https://user-images.githubusercontent.com/8518253/118883392-0a330900-b8aa-11eb-93b7-d3d98d9a0287.png)
Removed on next commit, thanks!